### PR TITLE
Adding restart prompt for plugins settings changes. (Refactoring included)

### PR DESCRIPTION
**Pull request by ryanml**
Fixes: #13574

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).
- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review). (Ask a Brave employee to help if you cannot access this document.)

## Test Plan:
 1. Visit `about:preferences#plugins`
 2. Toggle either the widevine or flash status
 3. Ensure that a restart notification is presented
 4. Restart
 5. Ensure that the changed settings persist
 6. Repeats steps 1 and 2
 7. Choose to not restart when the restart notification appears
 8. Manually restart the browser
 9. Ensure that the changed settings persist

## Reviewer Checklist:

- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review) if one was not already requested.

Tests

- [x] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

@diracdeltas this was a straightforward use of IPC to trigger a restart.

I felt compelled when implementing this to condense both the toggleWidevine and toggleFlash in to a general togglePlugin callback, in order to avoid duplicating code. This should make it easier and cleaner in the future when more plugins are integrated with Brave. Let me know your thoughts!

### CODE REVIEW

File: app/renderer/components/preferences/pluginsTab.js [hunk 1]
```diff
@@ -24,25 +26,35 @@ const {SettingsList, SettingCheckbox} = require('../common/settings')
...
+            temporary: false
+          })
+        }
+        break
     }
-  }
-
-  onToggleWidevine (e) {
-    aboutActions.setResourceEnabled(widevine, e.target.value)
+    ipc.send(messages.PREFS_RESTART)
```
Comment (diracdeltas) [inline app/renderer/components/preferences/pluginsTab.js]: i think it's slightly better to move this and the other `ipc.send(messages.PREFS_RESTART)` into an action in `js/about/aboutActions`

Comment (ryanml) [reply app/renderer/components/preferences/pluginsTab.js]: @diracdeltas sounds like a good idea. I can include that in to an action in this PR. Should I move all of the other restart notices in the codebase to the new aboutAction as well, or do that in a separate PR? Thanks!

Comment (diracdeltas) [reply app/renderer/components/preferences/pluginsTab.js]: @ryanml you can do it all in this PR, thanks

File: js/about/aboutActions.js [hunk 2]
```diff
@@ -102,6 +102,13 @@ const aboutActions = {
     ipc.send(messages.RESET_SYNC)
   },
 
+  /**
+   * Dispatches a message to trigger a browser restart notification
+   */
+  requireRestart: function () {
```
Comment (diracdeltas) [inline js/about/aboutActions.js]: this should send the key and value so the restart handler doesn't prompt for restart if you toggle the setting 2 times in a row. you can also replace PREFS_RESTART in ` js/about/preferences.js` with this.

**Comment by diracdeltas** [comment]
master / 0.24.x: 95bd9ab49186c85572676e50726a17994103d979

**Comment by ryanml** [comment]
Thanks for looking this over! 👍

**Pull request merged**